### PR TITLE
Lease to an online private runner by default; account default pool; runner field in the flow editor

### PR DESCRIPTION
**Pull request by d-mo**
Precedence: trigger override > flow.runner_pool > account.default_runner_pool > any online private runner > hosted. Adds the account setting (migration + API + Runners page control), the runner pool field and next-pool hint in the flow editor, and an explicit "server" sentinel. Lease and queue semantics unchanged. Backend and frontend unit tests.

> [!NOTE]
> **Low Risk**
> Changes the default execution target (hosted → online private runner) with a clear precedence chain and explicit "server" sentinel; the busy-vs-online edge case is resolved so a busy-only fleet now falls back to hosted compute instead of queuing to failure, and the path is well-tested.
>
> **Overview**
> Introduces an account-level default runner pool and a full precedence chain (trigger > flow > account default > auto > hosted) with an explicit "server" sentinel, exposing the setting via migration, API, and the Runners page, plus a runner-pool field with next-pool hint in the flow editor. Backend and frontend unit tests included.
>
> <sup>Written by [Preloop PR Reviewer](https://preloop.ai) for commit 5220c406. Updates automatically on new commits.</sup>

### CODE REVIEW

File: backend/preloop/services/runner_service.py [hunk 1]
```diff
@@ -35,20 +38,87 @@ def mint_runner_token() -> str:
...
+    if account_id is None:
+        return None
+    row = crud_account.get(db, id=account_id)
+    if row is None:
+        return None
+    value = getattr(row, "default_runner_pool", None)
+    return value if isinstance(value, str) or value is None else None
+
+
+def _has_online_private_runner(db: Session, account_id: Any) -> bool:
```
[the remark this answers is not in the document]
Comment (d-mo) [reply backend/preloop/services/runner_service.py:77]: Idle-only. `_has_online_private_runner` now requires `status == "online"` and no `pending_job`, so a busy-only fleet falls back to hosted compute instead of queuing to failure.

File: frontend/src/components/preloop-flow-form.ts [hunk 2]
```diff
@@ -657,6 +697,77 @@ export class PreloopFlowForm extends LitElement {
...
+  private normalizedFlowRunnerPool(): string | null {
+    const typed = (this.customRunnerPool || '').trim();
+    if (typed) {
+      return typed;
+    }
+    const selected = (this.flow.runner_pool || '').trim();
+    return selected || null;
+  }
+
+  private runnerPoolSelectValue(): string {
```
[the remark this answers is not in the document]
Comment (d-mo) [reply frontend/src/components/preloop-flow-form.ts:709]: Done. `runnerPoolSelectValue()` is now a single trim of `flow.runner_pool`; the unknown-value option still lives in `renderRunnerPoolField()`.

File: frontend/src/views/authed/runners-view.ts [hunk 3]
```diff
@@ -187,6 +216,49 @@ export class RunnersView extends LitElement {
...
+      this.defaultRunnerPool = updated.default_runner_pool ?? null;
+    } catch (err) {
+      this.defaultError =
+        err instanceof Error ? err.message : 'Failed to save default runner';
+    } finally {
+      this.savingDefault = false;
+    }
+  }
+
+  private renderDefaultPoolControl() {
```
[the remark this answers is not in the document]
Comment (d-mo) [reply frontend/src/views/authed/runners-view.ts:237]: Done. If the saved account default is missing from the online-runner options, it is pushed as an extra option so the select is not blank.
